### ISSUE TITLE
s22i doesn't send metrics

### ISSUE THREAD

**Issue by Rhic99o** [cagnulein/qdomyos-zwift#2883]
The bug is I don't know what I'm doing 😔 
 
So I've installed on my s22i using the browser

Installed the app on my android phone and have data going from the bike to my phone. 

Using zwift on a laptop they appear to be finding the app as a wahoo sensor 

But no data is transfering to zwift. So I pedal and data on phone but not in zwift. 

Desktop is windows.

**Comment by Rhic99o** [comment]
Have installed the android aps onto the phone running QZ but the files are empty with no bug reports. I have switched on bug reporting and recreated the issue. But no logs are appearing

**Comment by cagnulein** [comment]
Enable the android document setting in the experimental settings and then you will find them in the /documents/qz folder 

Let me know

**Comment by Rhic99o** [comment]
These are how I have the settings. Is that correct?

**Comment by Rhic99o** [comment]
Just wanted to check the bug reporting is turned on correctly.

**Comment by cagnulein** [comment]
I don't see any attachments, where did you upload it?

**Comment by Rhic99o** [comment]
I added as an attachment in the email. Can you see the attached log?

**Comment by Rhic99o** [comment]
Latest bug log where I replicated the issue

**Comment by cagnulein** [comment]
Nope you have to attach directly in the github web interface otherwise I can't see them

**Comment by Rhic99o** [comment]
[debug-Fri_Dec_6_15_47_44_2024.log](https://github.com/user-attachments/files/18041168/debug-Fri_Dec_6_15_47_44_2024.log)

**Comment by Rhic99o** [comment]
Hopefully that worked. Most of the files where to big for GitHub to let me upload

**Comment by cagnulein** [comment]
Yes that worked. If you went to upload bigger files zip it first

I checked your log and your bike is logging only speed and grade.

We have 2 possibilities : or I can derive wattage from speed and inclination or you can try this newer apk https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12175092604/artifacts/2277746109

That's reading the screen and it sends back the information to qz.

It will record the screen to do so 

Let me know!

**Comment by Rhic99o** [comment]
Thank you. I used the bowser on the bike to download from the tinyurl link when I set up. Do I need to connect to the bike over WiFi from my laptop to transfer this file? If so do I need to download other files as well from git hub to make this work?

**Comment by cagnulein** [comment]
Do you mean to download the apk? You have to use the same browser method that you did already

**Comment by Rhic99o** [comment]
Ok thanks

**Comment by Rhic99o** [comment]
So do I type that APK link into the browser on the bike?

**Comment by cagnulein** [comment]
Yes you need probably to login to github.com first

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 7 dic 2024 alle 12:54 Rhic99o ***@***.***> ha
scritto:

**Comment by Rhic99o** [comment]
Thanks I now have watts but not much else on the android qz app and nothing is feeding through to zwift. Will try and attach another log

**Comment by Rhic99o** [comment]
[Uploading debug-Sat_Dec_7_12_42_59_2024.log

**Comment by Rhic99o** [comment]
[debug-Sat_Dec_7_12_37_35_2024.log](https://github.com/user-attachments/files/18048400/debug-Sat_Dec_7_12_37_35_2024.log)

**Comment by cagnulein** [comment]
Ok that's great, I guess we are missing a small piece now.
Unfortunately the log is corrupted because you need to wait the upload is completed before pressing comment button
Let me know

**Comment by cagnulein** [comment]
Great! I will check it ASAP

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 7 dic 2024 alle 13:58 Rhic99o ***@***.***> ha
scritto:

**Comment by Rhic99o** [comment]
Thanks so much for your help, I really appreciate it.

**Comment by Rhic99o** [comment]
Also I forgot to ask using this method can I get the bike to automatically change incline and resistance?

**Comment by cagnulein** [comment]
Yes you can this as well!
I'm still out of town

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 8 dic 2024 alle 09:33 Rhic99o ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
hi @Rhic99o i checked your log. in order to add also the other metrics, i created a new apk that sends more information to the qz main app

https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12232123005/artifacts/2292532363

install it and then create me a debug log from qz as you did so i should have more info to work on them

Thanks!

**Comment by Rhic99o** [comment]
[debug-Wed_Dec_11_14_19_11_2024.log](https://github.com/user-attachments/files/18097296/debug-Wed_Dec_11_14_19_11_2024.log)

**Comment by Rhic99o** [comment]
Hi new debug attached. Not a lot of data showing on the qz app on my phone now.

**Comment by cagnulein** [comment]
@Rhic99o can you show me a screenshot of the manual windows of ifit on the bike's tablet? it seems that OCR is having hard time reading the metrics

**Comment by Rhic99o** [comment]
![17339293499772670870256153618304](https://github.com/user-attachments/assets/a6d39b03-6068-4094-ab8b-a59cf6b46714)

**Comment by cagnulein** [comment]
hah this this is ifit v2! ok could you please try this apk instead https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12272586209/artifacts/2304492203 ?

**Comment by Rhic99o** [comment]
[debug-Wed_Dec_11_15_47_28_2024.log](https://github.com/user-attachments/files/18098722/debug-Wed_Dec_11_15_47_28_2024.log)

**Comment by Rhic99o** [comment]
Thanks that's working a bit better with data. Connection to swift not quite there though

**Comment by cagnulein** [comment]
where are you using zwift? try to use zwift on the same android phone with qz but enable first the android notification icon in the experimental settings in qz
let me know!

**Comment by Rhic99o** [comment]
That seems to be working better using the android phone running zwift and qz. There is a huge lag between the bike screen and qz. Is there a way to improve this ? Also I don't think the auto incline and resistance are working. Thanks so much for all your help.

**Comment by Rhic99o** [comment]
[debug-Wed_Dec_11_16_19_59_2024.log](https://github.com/user-attachments/files/18099103/debug-Wed_Dec_11_16_19_59_2024.log)

**Comment by cagnulein** [comment]
yes auto incline needs to map the coordinates of the touchscreen. the delay is the biggest issue. how many seconds of delay do you have?

**Comment by Rhic99o** [comment]
Quiet long easily 10 seconds I would guess

**Comment by cagnulein** [comment]
This is the main issue so. The thing that we can do is using the phone as a
ocr server or a pc for this.
Do you have a pc?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 11 dic 2024 alle 17:53 Rhic99o ***@***.***> ha
scritto:

**Comment by Rhic99o** [comment]
Let's try the phone I guess

**Comment by cagnulein** [comment]
The pc solution is already ready, the phone solution is to create from
scratch. So if you have a pc a suggest to go on the pc way

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 11 dic 2024 alle 17:59 Rhic99o ***@***.***> ha
scritto:

**Comment by Rhic99o** [comment]
Ok can try my laptop

**Comment by cagnulein** [comment]
@victorypoint could you please append here the instructions to make the ocr on pc for the nordictrack bike with ifit v2?

Thanks!

**Comment by cagnulein** [comment]
@victorypoint Only the read out of the metrics from ifit2 on a bike

**Comment by victorypoint** [comment]
@cagnulein ok I just have to package it up and do final testing.

**Comment by cagnulein** [comment]
Ok in the meantime I will try to remove the debug log to check if I can speed it up

**Comment by Rhic99o** [comment]
Thanks

**Comment by cagnulein** [comment]
Also another idea is to remove the debug log from the companion and use
only the portion of the screen useful for the data

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 11 dic 2024 alle 22:31 Rhic99o ***@***.***> ha
scritto:
